tests/test_functions/cygwin_nosymlinks.fish

```diff
@@ -0,0 +1,11 @@
+function cygwin_nosymlinks --description \
+    "Return 0 if Cygwin fakes symlinks, return 1 otherwise"
+
+    if string match -q "MSYS*" (uname)
+        string match -qv "*winsymlinks*" "$MSYS"
```

Comment:
probably `not string match -q ...` would be easier to read than `string match -vq`.
(Also the if/else cascade could be a switch with `case 'MSYS*'`` etc. but either works for me)